swim/join_sender.go

```diff
@@ -112,6 +130,9 @@ func newJoinSender(node *Node, opts *joinOpts) (*joinSender, error) {
 		node: node,
 	}
 
+	// Parse bootstrap hosts into a map
```

Comment:
I haven't thought too much about how the concerns of a join sender would be separated. It /was/ clear to me that the fan-out sending/aggegration were one thing and who you're fanning-out to was another. But now, I'm not so sure that's true. Maybe they're more inter-related than I originally thought. I was thinking that a Selector abstraction would provide a sort of Iterator API like `NextGroup()` and the join sender could use it until "joinSize" were satisfied. You don't have to address these refactoring concerns now.